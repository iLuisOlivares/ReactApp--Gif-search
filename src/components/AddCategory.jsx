import React, { useState } from 'react';
import PropTypes from 'prop-types';

function AddCategory({ setCategories }) {

    const [inputValue, setInputValue] = useState("");


    const handleOnChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();

        console.log('HandleSubmit:', inputValue);
        if (inputValue.trim().length > 2) {
            setCategories(cats => [e.target[0].value, ...cats]);
            setInputValue('');
        }

    }

    return (
        <form className='search' onSubmit={handleSubmit}>
            <input

                type="text"
                value={inputValue}
                onChange={handleOnChange}
            ></input>
        </form>);

}

export default AddCategory;

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}