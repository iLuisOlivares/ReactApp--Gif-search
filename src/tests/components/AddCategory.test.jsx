import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Add category testing', () => {

    const setCategories = jest.fn();

    let wrapper = shallow(<AddCategory setCategories={setCategories}></AddCategory>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}></AddCategory>);
    });

    test('Debe mostrar el snapshot', () => {
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe cambiar el input', () => {
        const value = 'Ola mundo';
        wrapper.find('input').simulate('change', { target: { value } });

    });

    test('Debe Simular el envio del formulario', () => {
        const form = wrapper.find('form');

        form.simulate('submit', { preventDefault: () => { } });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Llamar a setCategories y limpiar la caja de texto', () => {

        const value = 'Naruto';

        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault: () => { } });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        expect(wrapper.find('input').prop('value')).toBe('');
    });

});
