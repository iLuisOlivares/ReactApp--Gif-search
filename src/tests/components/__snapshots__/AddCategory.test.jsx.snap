// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add category testing Debe mostrar el snapshot 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
